src/test/java/io/thinkit/edc/client/connector/ApplicationObservabilityTest.java

```diff
@@ -37,6 +38,28 @@ void should_check_health() {
         });
     }
 
+    @Test
+    void should_check_health_async() {
```

Comment:
to clearly separate async and sync tests there's a junit trick:

create two inner classes into the test class, one called `Sync` and the other `Async`, and mark them with the `@Nested` annotation